### ISSUE TITLE
`blanks-around-lists` (`MD032`) - false positive on list followed by indented list when the previous item in the indented list ends with a `.`

### ISSUE THREAD

**Issue by DetachHead** [rvben/rumdl#723]
### rumdl version

0.2.31

### Installation method

Other

### Bug description

when a list is followed by an indented block that contains a list, and that second indented list contains an item that ends with a `.`, the subsequent item in the list incorrectly causes `MD032`

### Steps to reproduce

```md
1. no error here

!!! example

    1. no error here.
    2. error here because previous line ends with a "."
    3. no error here
```

### Expected behavior

no errors

### Actual behavior

```
asdf.md:6:1: [MD032] Ordered list starting with non-1 should be preceded by blank line [*]
```

### Configuration

```toml
[global]
flavor = "mkdocs"
enable = ["MD032"]
```

### Operating system

linux

**Comment by rvben** [comment]
Fixed in v0.2.32. Under the mkdocs flavor, admonition-nested lists have stale list metadata internally, so MD032's fallback for non-1 ordered items judged them by the preceding line's trailing punctuation. The fallback now recognizes a nested list (an ordered marker line at the same indent above) and exempts it, while prose followed by a non-1 item still warns inside admonitions exactly as outside them. Thanks for the report.
